internal/integrations/github/github.go

```diff
@@ -0,0 +1,53 @@
+package github
+
+import (
+	"context"
+	"errors"
+
+	"github.com/google/go-github/v26/github"
+	"golang.org/x/oauth2"
+)
+
+var errTravisCIBuildNotFound = errors.New("could not find a TravisCI build for specified ref")
+
+// ChecksService wraps github.ChecksService
+type ChecksService interface {
+	ListCheckRunsForRef(context.Context, string, string, string, *github.ListCheckRunsOptions) (*github.ListCheckRunsResults, *github.Response, error)
+}
+
+// Github handles integrations with the Github API
+type Github struct {
+	Org    string
+	Client *github.Client
+	Checks ChecksService
+}
+
+// New creates a new GitHub integration
+func New(ctx context.Context, org string, accessToken string) *Github {
```

Comment:
It would make testing slightly more straightforward if this constructor func depended directly on the `ChecksService` interface, ie. `New(checks ChecksService, org string) *Github`. 

The downside is that the caller now needs to know about the oauth and github pkgs so it can provide a client. I'm not sure whether slightly cleaner tests are worth losing that encapsulation. It might become easier to decide as more things are added to this github pkg, so I'd be okay punting on it.

Reply:
Yeah I considered doing that but I want to keep those details hidden for now.